eq-author-api/schema/resolvers/base.js

```diff
@@ -701,6 +719,141 @@ const Resolvers = {
       await saveQuestionnaire(ctx.questionnaire);
       return ctx.questionnaire;
     },
+    createComment: async (_, { input }, ctx) => {
+      const { componentId } = input;
+      const questionnaire = ctx.questionnaire;
+      const questionnaireComments = await getCommentsForQuestionnaire(
+        questionnaire.id
+      );
+      const newComment = {
+        id: uuid.v4(),
+        commentText: input.commentText,
+        userId: ctx.user.id,
+        createdTime: new Date(),
+        replies: [],
+      };
+
+      const componentComments = questionnaireComments.comments[componentId];
+
+      if (componentComments) {
+        questionnaireComments.comments[componentId].push(newComment);
+      } else {
+        questionnaireComments.comments[componentId] = [newComment];
+      }
+
+      await saveComments(questionnaireComments);
+      publishCommentUpdates(componentId);
+
+      return newComment;
+    },
+    deleteComment: async (_, { input }, ctx) => {
+      const { componentId } = input;
+      const questionnaire = ctx.questionnaire;
+      const questionnaireComments = await getCommentsForQuestionnaire(
+        questionnaire.id
+      );
+
+      const componentComments = questionnaireComments.comments[componentId];
+      if (componentComments) {
+        remove(componentComments, ({ id }) => id === input.commentId);
+        await saveComments(questionnaireComments);
+      }
+      publishCommentUpdates(componentId);
+      return componentComments;
+    },
+    updateComment: async (_, { input }, ctx) => {
+      const { componentId } = input;
+      const questionnaire = ctx.questionnaire;
+      const questionnaireComments = await getCommentsForQuestionnaire(
+        questionnaire.id
+      );
+      const pageComments = questionnaireComments.comments[componentId];
+
+      if (!pageComments) {
+        throw new Error("No comments found");
+      }
+
+      const commentToEdit = pageComments.find(
+        ({ id }) => id === input.commentId
+      );
+      commentToEdit.commentText = input.commentText;
+      commentToEdit.editedTime = new Date();
+      await saveComments(questionnaireComments);
+
+      publishCommentUpdates(componentId);
+
+      return commentToEdit;
+    },
+    createReply: async (_, { input }, ctx) => {
+      const { componentId } = input;
+      const questionnaire = ctx.questionnaire;
+      const questionnaireComments = await getCommentsForQuestionnaire(
+        questionnaire.id
+      );
+
+      const newReply = {
+        id: uuid.v4(),
+        parentCommentId: input.commentId,
+        commentText: input.commentText,
+        userId: ctx.user.id,
+        createdTime: new Date(),
+      };
+      let parentComment = questionnaireComments.comments[componentId].find(
+        ({ id }) => id === input.commentId
```

Comment:
If you do the above, this would become

```suggestion
        ({ id }) => id === commentId
```